modules/oe_theme_content_organisation/src/Plugin/ExtraField/Display/TeaserDetailsExtraField.php

```diff
@@ -85,32 +79,38 @@ public function viewElements(ContentEntityInterface $entity) {
       ],
     ];
     $cache = CacheableMetadata::createFromRenderArray($build);
+    $items = [];
 
-    $contact_access = $contact->access('view', NULL, TRUE);
-    $cache->addCacheableDependency($contact);
-    $cache->addCacheableDependency($contact_access);
+    /** @var \Drupal\oe_content_entity_contact\Entity\ContactInterface $contact */
+    foreach ($entity->get('oe_organisation_contact')->referencedEntities() as $contact) {
+      if (!$contact) {
+        continue;
+      }
+      $contact_access = $contact->access('view', NULL, TRUE);
+      $cache->addCacheableDependency($contact);
+      $cache->addCacheableDependency($contact_access);
 
-    if (!$contact_access->isAllowed()) {
-      $cache->applyTo($build);
-      return $build;
-    }
+      if (!$contact_access->isAllowed()) {
+        $cache->applyTo($build);
+        return $build;
+      }
 
-    $items = [];
-    $fields = [
-      'oe_website' => [],
-      'oe_email' => ['type' => 'email_mailto'],
-      'oe_phone' => [],
-      'oe_address' => [
-        'type' => 'oe_theme_helper_address_inline',
-        'settings' => ['delimiter' => ', '],
-      ],
-    ];
-    foreach ($fields as $field_name => $display_options) {
-      if (!$contact->get($field_name)->isEmpty()) {
-        $items[] = $this->getRenderableFieldListItem($contact, $field_name, $display_options);
+      $fields = [
+        'oe_website' => [],
+        'oe_email' => ['type' => 'email_mailto'],
+        'oe_phone' => [],
+        'oe_address' => [
+          'type' => 'oe_theme_helper_address_inline',
+          'settings' => ['delimiter' => ', '],
+        ],
+      ];
+      foreach ($fields as $field_name => $display_options) {
+        if (!$contact->get($field_name)->isEmpty()) {
+          $items[] = $this->getRenderableFieldListItem($contact, $field_name, $display_options);
```

Comment:
You are merging here the fields of all the contacts and this is not what needs to happen if you look at the screenshot. 

I have two contacts:

![Screen Shot 2021-04-20 at 09 28 53](https://user-images.githubusercontent.com/5848933/115348082-7985e200-a1b2-11eb-9496-a4bc6cf36e5e.png)

And this is the resulting teaser:

![Screen Shot 2021-04-20 at 09 29 19](https://user-images.githubusercontent.com/5848933/115348127-87d3fe00-a1b2-11eb-92a2-f5ade5f9cf17.png)

Compare please to the screenshot from the ticket.